Arm/Insts/Common.lean

```diff
@@ -93,6 +156,13 @@ def Aligned (x : BitVec n) (a : Nat) : Prop :=
 instance : Decidable (Aligned x a) := by
   cases a <;> simp [Aligned] <;> infer_instance
 
+theorem Aligned_BitVecSub_64_4 {x : BitVec 64} {y : BitVec 64}
```

Comment:
Let's add the `state_simp_rules` tag to the three `Aligned`-related theorems.

Reply:
I am not convinced that these are good `simp` rues, because as far as I can tell, we will almost never never naturally have the proof that `x` and `y` are aligned! This will almost always require proof search, as far as I can tell?

Reply:
Aha, good point! We only use `Aligned` in `CheckSPAlignment` right now, and since we're going to have automation for the latter, I rescind my comment.